packages/replay-next/components/sources/utils/comments.ts

```diff
@@ -89,26 +89,26 @@ export async function createTypeDataForSourceCodeComment(
   // Secondary label is used to store the syntax-highlighted markup for the line
   const streamingSource = await getStreamingSourceContentsAsync(replayClient, sourceId);
   if (streamingSource != null) {
-    const parsedSource = await streamingSyntaxParsingCache.readAsync(streamingSource, fileName);
+    const parsedSource = parseStreaming(streamingSource, fileName);
     if (parsedSource != null) {
-      if (parsedSource.rawTextByLine.length < lineNumber) {
+      if (parsedSource.data?.text.length ?? 0 < lineNumber) {
         // If the streaming source hasn't finished loading yet, wait for it to load;
         // Note that it's important to check raw lines as parsed lines may be clipped
         // if the source is larger than the parser has been configured to handle.
         await new Promise<void>(resolve => {
           parsedSource.subscribe(() => {
-            if (parsedSource.rawTextByLine.length >= lineNumber) {
+            if (parsedSource.data?.text.length ?? 0 >= lineNumber) {
               resolve();
             }
           });
         });
       }
 
-      rawText = parsedSource.rawTextByLine[lineNumber - 1];
+      rawText = parsedSource.data!.text[lineNumber - 1];
       if (rawText.length <= maxTextLength) {
         // If the raw text is longer than the max length, we can't safely use the parsed tokens.
-        if (parsedSource.parsedTokensByLine.length >= lineNumber) {
-          parsedTokens = parsedSource.parsedTokensByLine[lineNumber - 1] ?? null;
+        if (parsedSource.value?.length ?? 0 >= lineNumber) {
+          parsedTokens = parsedSource.value?.[lineNumber - 1] ?? null;
```

Comment:
I think we don't need to still be using `!` and `?` at this point. We should be able to add a TypeScript assertion after  line 105.